std/net/curl.d

```diff
@@ -627,6 +627,60 @@ unittest
     assert(res == cast(ubyte[])[17, 27, 35, 41]);
 }
 
+/* Encode associative array using www-form-urlencoding
+ * perhaps this should live in std.uri
+ */
+private string urlEncode()(in string[string] values)
+{
+    bool first = true;
+    Appender!string enc;
+    enc.reserve(values.length * 512);
+    foreach (k, v; values)
+    {
+        if (!first)
+            enc ~= '&';
+        import std.uri : encodeComponent;
+
+        enc ~= encodeComponent(k);
+        enc ~= "=";
+        enc ~= encodeComponent(v);
+        first = false;
+    }
+    return enc.data;
+}
+
+/** HTTP post content.
+ *
+ * Params:
+ * url = resource to post to
+ * postData = data to send as the body of the request. An associative array of $(D string) is accepted and will be
+ *            encoded using www-form-urlencoding
+ * conn = HTTP connection to use
+ *
+ * Returns:
+ * A $(D char[]) containing the content of the resource pointed to by the URL.
```

Comment:
nitpick: we usually indent with four spaces the following lines of `Params` and `Returns`. Not all parts of Phobos might stick to this, but you are cordially invited to fix this when you browse code ;-)